Philips.CodeAnalysis.Test/Maintainability/Maintainability/AvoidAssignmentInConditionCodeFixProviderTest.cs

```diff
@@ -0,0 +1,84 @@
+﻿// © 2023 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Philips.CodeAnalysis.MaintainabilityAnalyzers.Maintainability;
+using Philips.CodeAnalysis.Test.Helpers;
+using Philips.CodeAnalysis.Test.Verifiers;
+
+namespace Philips.CodeAnalysis.Test.Maintainability.Maintainability
+{
+	/// <summary>
+	/// Test class for <see cref="AvoidAssignmentInConditionCodeFixProvider"/>.
+	/// </summary>
+	[TestClass]
+	public class AvoidAssignmentInConditionCodeFixProviderTest : CodeFixVerifier
+	{
+		private const string SimpleAssignmentViolation = @"
+namespace AssignmentInConditionUnitTests {
+	public class Program {
+		public bool Main() {
+			bool flag = false;
+			if (flag = true) {
+				// Do nothing
+			}
+		}
+	}
+}";
+
+		private const string SimpleAssignmentFixed = @"
+namespace AssignmentInConditionUnitTests {
+	public class Program {
+		public bool Main() {
+			bool flag = false;
+			flag = true;
+			if (flag) {
+				// Do nothing
+			}
+		}
+	}
+}";
+
+		private const string TernaryViolation = @"
+namespace AssignmentInConditionUnitTests {
+	public class Program {
+		public bool Main() {
+			bool flag = false;
+			int result = (flag = true) ? 10 : 20;
+		}
+	}
+}";
+
+		private const string TernaryFixed = @"
+namespace AssignmentInConditionUnitTests {
+	public class Program {
+		public bool Main() {
+			bool flag = false;
+			flag = true;
+			int result = (flag) ? 10 : 20;
+		}
+	}
+}";
+
+		[DataTestMethod]
```

Comment:
As I've attempted to create a fixer for this Analyzer before, I can tell from experience that there are more complex scenarios that we need to test, like:

- Assigning the return value of a method call
- Assigning a property, which is fine as-is and shall NOT be flagged
- Linq statements like `Any()` are probably OK to keep as-is ??